src/Apps/Legion/Components/WizardShell/Internal/DataSourceStep/DataSourceStep.tsx

```diff
@@ -273,6 +286,19 @@ const DataSourceStep: React.FC<IDataSourceStepProps> = (props) => {
         }
     }, [createDatabaseState?.adapterResult]);
 
+    // effects
+    useEffect(() => {
```

Comment:
can we pull this up to the context and use an action to `registerPrimaryAction` instead?